### PR TITLE
add emacs snapshot update action

### LINKED ISSUE #21: Create a scheduled action to update Emacs snapshots for CI
Based on a discussion in https://github.com/emacs-php/php-ts-mode/pull/20#issuecomment-1574809973 we will need to periodically refresh dependencies to be able to test with the latest Emacs snapshots.

We can create something similar to https://github.com/purcell/nix-emacs-ci/blob/master/.github/workflows/update-snapshot.yml

To update `flake.lock` we have to run:

```
nix flake update
```

### PR DESCRIPTION
**Pull request by piotrkwiecinski**
This is a scheduled action. It triggers once a month.
It updates flake lock and creates a PR if file content changed.

In addition it's possible to trigger the action manually.

resolves #21

### CODE REVIEW

**Comment by piotrkwiecinski** [comment]
@akirak would could you have a look and validate if we need to add anything else? You're more experienced with nix.
Should we include:
```
extraPullNames: nix-community
```

File: .github/workflows/update-snapshot.yml [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@v3
+    - uses: cachix/install-nix-action@v21
+      with:
+        github_access_token: ${{ secrets.GITHUB_TOKEN }}
+        nix_path: nixpkgs=channel:nixos-unstable
```
Comment (akirak) [inline .github/workflows/update-snapshot.yml]: We don't need this line because we only run `nix flake update` which doesn't depend on a nixpkgs channel.

**Comment by akirak** [comment]
> Should we include: extraPullNames: nix-community

No. `nix flake update` only updates the inputs specified in the `inputs` section of `flake.nix`, so we won't build packages in this workflow. (`flake-utils` and `systems` are not defined in the flake, but they are included in the global registry of Nix.)

Even if we build packages, we only need Emacs executables, which is available on `emacs-ci` cache on Cachix, which is configured in the `flake.nix` of this repository. To enable the cache, you only need `--accept-flake-config` flag (or `accept-flake-config = true` in the config file as in `test` workflow):

```sh
nix develop .#emacs-release-snapshot --accept-flake-config
```

**Review by akirak — COMMENTED** [review]
Note that peter-evans/create-pull-request doesn't trigger workflows by default. See https://github.com/peter-evans/create-pull-request/blob/main/docs/concepts-guidelines.md#triggering-further-workflow-runs for information.

**Comment by piotrkwiecinski** [comment]
@akirak do we also need `--accept-flake-config` added to flake update action?

**Comment by piotrkwiecinski** [comment]
> Note that peter-evans/create-pull-request doesn't trigger workflows by default. See https://github.com/peter-evans/create-pull-request/blob/main/docs/concepts-guidelines.md#triggering-further-workflow-runs for information.

Thank you for highlighting it. 
@zonuexe which option do you prefer?
I reckon we could go with https://github.com/peter-evans/create-pull-request/blob/main/docs/concepts-guidelines.md#authenticating-with-github-app-generated-tokens or https://github.com/peter-evans/create-pull-request/blob/main/docs/concepts-guidelines.md#push-using-ssh-deploy-keys but it's going to need some ground work done by you.

As we don't push to remote repository deployment ssh key should be enough.

**Comment by akirak** [comment]
> do we also need --accept-flake-config added to flake update action?

No, we need it only for `nix develop`. I have already added it to the `test` workflow (see below), but it's unneeded for updating.

```yaml
    - uses: cachix/install-nix-action@v21
      with:
        extra_nix_config: |
          access-tokens = github.com=${{ secrets.GITHUB_TOKEN }}
          # Use the binary cache configured in `nixConfig` section
          accept-flake-config = true
```

**Comment by piotrkwiecinski** [comment]
@zonuexe I have updated PR to use https://github.com/peter-evans/create-pull-request/blob/main/docs/concepts-guidelines.md#push-using-ssh-deploy-keys

Please generated a SSH key and add private key as `SSH_PRIVATE_KEY` in GitHub action secrets.

This should be good enough for us.

**Comment by zonuexe** [comment]
I generated a new SSH key and set it as deploy key and `secrets.SSH_PRIVATE_KEY`.

**Comment by zonuexe** [comment]
Thank you all.

**Pull request merged**